authors/apps/social_auth/login_register.py

```diff
@@ -0,0 +1,32 @@
+from django.core.exceptions import ValidationError
+
+from authors.apps.authentication.models import User
+
+
+def login_or_register_social_user(social_user):
+    try:
+        user = User.objects.get(email=social_user.get('email'))
+        return {
+            "email": user.email,
+            "username": user.username,
+            "token": user.auth_token()
+        }
+    except User.DoesNotExist:
+        if social_user.get("email") is None:
+            raise ValidationError("Users must have an email address.")
+
+        new_social_user = {
+            "email": social_user.get("email"),
+            "username": social_user.get("name", "unknown"),
```

Comment:
@richien, what would happen if more than one user signing up has no username field yet it is a unique field.
I suggest you should find a way to create a unique username which they can edit later

Reply:
Thank you for the feedback. This has been implemented.